core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1265,6 +1257,12 @@ void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, ExecutionE
                         canProcess = false;
                         continue;
                     }
+                    // This works as retry mechanism for failed deletion remote segments. Rather than waiting for the 
```

Comment:
Is the comment referring to dangling segments? Can we update it to:

```
This works as retry mechanism for failed deletion remote segments.
```
to 

```
This works as retry mechanism for dangling remote segments that are failed to delete in previous attempt. ...
```

Reply:
> Is the comment referring to dangling segments?

Yes, once the broker fails the deletion, the segment to be deleted can be considered dangling. This can happen after a failed copy or when running the expiration logic.